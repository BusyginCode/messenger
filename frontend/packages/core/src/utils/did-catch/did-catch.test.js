import React from 'react'
import { mount } from 'enzyme'
import didCatch from './did-catch'

describe('core/utils/did-catch', () => {
  it('should render correctly', () => {
    const Child = () => <span>1</span>
    const Component = didCatch()(Child)
    expect(mount(<Component />)).toMatchSnapshot()
  })

  it('should render correctly with Error', () => {
    const Child = a => <span>{a.a.a}</span>
    const onError = jest.fn()
    const Component = didCatch({ onError })(Child)

    expect(mount(<Component />)).toMatchSnapshot()

    expect(onError).toHaveBeenCalledTimes(1)
  })

  it('should render correctly with Error & Component', () => {
    const Child = a => <span>{a.a.a}</span>
    const ErrorComponent = () => <span>error component</span>
    const Component = didCatch({ ErrorComponent })(Child)

    expect(mount(<Component />)).toMatchSnapshot()
  })
})
