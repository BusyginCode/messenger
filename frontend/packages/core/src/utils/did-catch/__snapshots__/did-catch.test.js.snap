// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core/utils/did-catch should render correctly 1`] = `
<didCatch(Child)>
  <Child>
    <span>
      1
    </span>
  </Child>
</didCatch(Child)>
`;

exports[`core/utils/did-catch should render correctly with Error & Component 1`] = `
<didCatch(Child)>
  <ErrorComponent
    error="TypeError: Cannot read property 'a' of undefined"
  >
    <span>
      error component
    </span>
  </ErrorComponent>
</didCatch(Child)>
`;

exports[`core/utils/did-catch should render correctly with Error 1`] = `<didCatch(Child) />`;
